### ISSUE TITLE
Custom Fonts settings disappear from controls when a custom theme is set in project settings

### ISSUE THREAD

**Issue by drentsoft** [godotengine/godot#18741]
**Godot version:**

Godot 3.0.2 stable (non-mono)

**OS/device including version:**

Windows 10 Pro x64

**Issue description:**

When setting a custom theme in project settings most text-based controls lose Custom Fonts settings in the inspector. Link Button seems to lose custom styles, colours and constants as well. Settings will only reappear once the theme is removed from project settings and the project is reloaded.
I believe this is related but distinct to #6098 as the issue only rectifies on project reload rather than selecting a different node

**Steps to reproduce:**
Create a scene
Add controls (Button, Link Button etc)
Create a new theme resource
Edit Theme > Create from Current Editor Theme
Save Theme resource
Project Settings > GUI > Theme > Set custom theme to newly created theme
Save scene
Select a control and settings should appear in the inspector as expected
Close the project
Re-open the project, open the scene.
Select a control, custom fonts settings should now be missing
Run project/scene, 2 errors should appear in the debugger

**Minimal reproduction project:**

[customThemeBug.zip](https://github.com/godotengine/godot/files/1987444/customThemeBug.zip)

**Comment by volzhs** [comment]
This looks similar with #7001 and it's fixed by 88bfb27abf35b347b4fdb0def15b340f33c89209
but this one still valid.

I tested by creating new theme but not with **Create from Current Editor Theme**
the new theme contains only custom font for label.
and set it as custom theme at project setting.
after restarting editor, `Control`s has custom fonts setting, **ERROR** for other gui texture are shown though.

so, it seems there is a problem with the theme created by **Create from Current Editor Theme**

**Comment by nightblade9** [comment]
This is exactly the same problem I posted about on Twitter: https://twitter.com/nightblade99/status/1098791315922243585

Repro steps:
- Use "Create from Current Editor Theme" to create a theme
- Save the theme and set it as your `Custom` theme for the project
- Try to edit a label to change fonts. `custom_fonts` is missing.

This is with 3.0.6.

Did you guys ever find a work-around for this? I tried removing the theme, but never thought to reload the project after.
